wave_matching/src/icp.cpp

```diff
@@ -13,17 +13,20 @@ ICPMatcher::ICPMatcher(float res) {
     } else {
         this->resolution = -1;
     }
-    wave::ConfigParser parser;
+    ConfigParser parser;
```

Comment:
Voxel resolution is already set via the constructor, the rest are parameters like maximum iterations, stopping criteria, maximum correspondences, etc. Are these really the type of parameters you would want to change between scan matches?
Multiple config files might be better for something like multi-scale optimisation because they expose all the configuration parameters without having to look at code.